### PR TITLE
mzIdentML Crosslinking Terms

### PR DESCRIPTION
**Pull request by mobiusklein**
This PR introduces the new terms for the mzIdentML extension for crosslinking.

@colin-combe
@lutzfischer

### CODE REVIEW

**Comment by javizca** [comment]
I checked the list of new terms before it was submitted, and it is fine for me

**Comment by colin-combe** [comment]
Thanks Josh!
Assuming I am looking at the right version of file, then some CV terms (MS:1002494, MS:1002508, MS:1002510, MS:1002511,  MS:1002512) still have the hyphenated "cross-link" in their defs? 

MS:1003023 also has it in it's def but as this is someone else's description of their software I think leave it as it is.
Cheers,
Colin

File: psi-ms.obo [hunk 1]
```diff
@@ -17521,16 +17527,18 @@ relationship: has_value_type xsd:double ! The allowed value-type for this CV ter
...
+is_a: MS:1002675 ! crosslinking result details
+relationship: has_regexp MS:1002665 ! regular expression for interaction scores derived from crosslinking
 relationship: has_value_type xsd:string ! The allowed value-type for this CV term
 
 [Term]
 id: MS:1002665
-name: regular expression for interaction scores derived from cross-linking
+name: regular expression for interaction scores derived from crosslinking
 def: "((?<int_ID>[0-9]+)[.](?<SUFFIX>[a|b]):(?<POS>[0-9]+|null):(?<SCORE_OR_VALUE>[-+]?[0-9]+(?:[.][0-9]+)?(?:[Ee][-+]?[0-9]+)?):(?<PASS_THRESHOLD>true|false))|(?<SCORE_ONLY>[-+]?[0-9]+(?:[.][0-9]+)?(?:[Ee][-+]?[0-9]+)?)" [PSI:PI]
+synonym: "regular expression for interaction scores derived from crosslinking" EXACT []
```
Comment (edeutsch) [inline psi-ms.obo:17541]: this is probably intended to have a hyphen in it to acknowledge the original name, as done with other terms?

File: psi-ms.obo [hunk 2]
```diff
@@ -21888,25 +21898,162 @@ name: Lipid shorthand identification confidence - Complete structure
...
-relationship: has_metric_category MS:4000008 ! ID based metric 
-relationship: has_value_type xsd:int ! The allowed value-type for this CV term 
-relationship: has_units UO:0000189 ! count unit 
+relationship: has_metric_category MS:4000008 ! ID based metric
+relationship: has_value_type xsd:int ! The allowed value-type for this CV term
+relationship: has_units UO:0000189 ! count unit
+
+[Term]
+id: MS:1003329
+name: loop-link spectrum identification item
```
Comment (edeutsch) [inline psi-ms.obo:21923]: If we're scrubbing out the hyphens in crosslink, should they also be scrubbed from looplink?

File: psi-ms.obo [hunk 3]
```diff
@@ -17521,16 +17527,18 @@ relationship: has_value_type xsd:double ! The allowed value-type for this CV ter
 
 [Term]
 id: MS:1002664
-name: interaction score derived from cross-linking
-def: "Parent term for interaction scores derived from cross-linking." [PSI:PI]
-is_a: MS:1002675 ! cross-linking result details
-relationship: has_regexp MS:1002665 ! regular expression for interaction scores derived from cross-linking
+name: interaction score derived from crosslinking
+def: "Parent term for interaction scores derived from crosslinking." [PSI:PI]
+synonym: "interaction score derived from crosslinking" EXACT []
```
Comment (edeutsch) [inline psi-ms.obo:17532]: this synonym should retain its hyphen?

**Review by edeutsch — COMMENTED** [review]
I think I spied a few errors, perhaps I'm mistaken, please address in-line comments. This is a pretty complex PR, so after addressing my comments, I think it would be good to get some more eyes and approvals on this.

File: psi-ms.obo (hunk 1 above)
Comment (colin-combe) [reply psi-ms.obo:17541]: yep, think so

File: psi-ms.obo (hunk 3 above)
Comment (colin-combe) [reply psi-ms.obo:17532]: yep, think so

File: psi-ms.obo (hunk 2 above)
Comment (colin-combe) [reply psi-ms.obo:21923]: well, this is a new term so its not being "scrubbed". 
But yes, Alexander Lietner proposed that is inconsistent to to have "loop-link" hyphenated when crosslink is not.
But then Lutz and I decided to have it hyphenated still, for no better reason than we felt it looked too weird without the hyphen. 
So, the hyphen currently present in "loop-link" is deliberate but needs further discussion. I can see Eric's/Alexander's point.

**Review by edeutsch — APPROVED** [review]

**Review by ypriverol — APPROVED** [review]

**Review by colin-combe — APPROVED** [review]

**Pull request merged**